docs/resources/mapreduce_cluster.md

```diff
@@ -0,0 +1,354 @@
+    data_volume_count = 1
+  }
+  analysis_core_nodes {
+    flavor = "c6.2xlarge.4.linux.bigdata"
+    node_number = 2
+    root_volume_type  = "SAS"
+    root_volume_size  = 300
+    data_volume_type  = "SAS"
+    data_volume_size   = 480
+    data_volume_count = 1
+  }
+  streaming_core_nodes {
+    flavor = "c6.2xlarge.4.linux.bigdata"
+    node_number = 2
+    root_volume_type  = "SAS"
+    root_volume_size  = 300
+    data_volume_type  = "SAS"
+    data_volume_size   = 480
+    data_volume_count = 1
+  }
+  analysis_task_nodes {
+    flavor = "c6.2xlarge.4.linux.bigdata"
+    node_number = 1
+    root_volume_type  = "SAS"
+    root_volume_size  = 300
+    data_volume_type  = "SAS"
+    data_volume_size   = 480
+    data_volume_count = 1
+  }
+  streaming_task_nodes {
+    flavor = "c6.2xlarge.4.linux.bigdata"
+    node_number = 1
+    root_volume_type  = "SAS"
+    root_volume_size  = 300
+    data_volume_type  = "SAS"
+    data_volume_size   = 480
+    data_volume_count = 1
+  }
+
+  tags = {
+    foo = "bar"
+    key = "value"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) The region in which to create the MapReduce cluster resource.
+  If omitted, the provider-level region will be used.
+  Changing this will create a new MapReduce cluster resource.
+
+* `availability_zone` - (Required, String, ForceNew) Specifies the availability zone in which to create
+  the cluster. Please following [reference](https://developer.huaweicloud.com/en-us/endpoint/?ECS)
+  Changing this will create a new MapReduce cluster resource.
+
+* `name` - (Required, String, ForceNew) Specifies the name of the MapReduce cluster.
+  The name can contain 2 to 64 characters, which may consist of letters, digits,
+  underscores (_) and hyphens (-).
+  Changing this will create a new MapReduce cluster resource.
+
+* `version` - (Required, String, ForceNew) Specifies the MapReduce cluster version.
+  Changing this will create a new MapReduce cluster resource.
+
+* `component_list` - (Required, List, ForceNew) Specifies the list of component names. For the components supported by
+  the cluster, please following [reference](https://support.huaweicloud.com/intl/en-us/productdesc-mrs/mrs_08_0005.html)
+  Changing this will create a new MapReduce cluster resource.
+
+* `master_nodes` - (Required, List, ForceNew) Specifies a list of the informations about the master nodes in the MapReduce cluster.
+  The `nodes` object structure of the `master_nodes` is documented below.
+  Changing this will create a new MapReduce cluster resource.
+
+* `vpc_id` - (Required, String, ForceNew) Specifies the VPC ID.
+  Changing this will create a new MapReduce cluster resource.
+
+* `subnet_id` - (Required, String, ForceNew) Specifies the network ID of a subnet.
+  Changing this will create a new MapReduce cluster resource.
+
+  -> **NOTE:** The duplication of subnet CIDR will cause the MRS cluster creation to fail.
```

Comment:
can we create a subent with duplication CIDR?

Reply:
the scenario that caused the failure is more complicated and the description has been removed.